python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -505,6 +505,28 @@ abstract class FunctionValue extends CallableValue {
     predicate isOverriddenMethod() {
         exists(Value f | f.overrides(this))
     }
+    
+    /** Whether `name` is a legal argument name for this function */
+    bindingset[name]
+    predicate isLegalArgumentName(string name) {
+        this.getScope().getAnArg().asName().getId() = name
+        or
+        this.getScope().getAKeywordOnlyArg().getId() = name
+        or
+        this.getScope().hasKwArg()
+    }
+    
+    /** Whether this is a "normal" method, that is, it is exists as a class attribute 
+     *  which is not wrapped and not the __new__ method. */
```

Comment:
```suggestion
     *  which is not a lambda and not the __new__ method. */
```

Reply:
This is a better reflection of what the underlying predicate does, so I would encourage accepting this suggestion.

I think we'll want to revisit this issue of whether excluding lambdas is actually useful, but I don't think this should be a priority. I'll create an issue for this and other oddities to look into when we get the time/urge.

Reply:
Couldn't find your issue, so I created it at https://github.com/github/codeql-python-team/issues/45